proto/cometbft/abci/v3/types.proto

```diff
@@ -0,0 +1,288 @@
+  rpc ApplySnapshotChunk(v1.RequestApplySnapshotChunk)
+      returns (v1.ResponseApplySnapshotChunk);
+  rpc PrepareProposal(v3.RequestPrepareProposal) returns (v2.ResponsePrepareProposal);
+  rpc ProcessProposal(RequestProcessProposal) returns (v2.ResponseProcessProposal);
+  rpc ExtendVote(RequestExtendVote) returns (ResponseExtendVote);
+  rpc VerifyVoteExtension(RequestVerifyVoteExtension) returns (ResponseVerifyVoteExtension);
+  rpc FinalizeBlock(RequestFinalizeBlock) returns (ResponseFinalizeBlock);
+}
+
+//----------------------------------------
+// Request types
+
+message Request {
+  oneof value {
+    v1.RequestEcho                echo                  = 1;
+    v1.RequestFlush               flush                 = 2;
+    v2.RequestInfo                info                  = 3;
+    RequestInitChain              init_chain            = 5;
+    v1.RequestQuery               query                 = 6;
+    v1.RequestCheckTx             check_tx              = 8;
+    v1.RequestCommit              commit                = 11;
+    v1.RequestListSnapshots       list_snapshots        = 12;
+    v1.RequestOfferSnapshot       offer_snapshot        = 13;
+    v1.RequestLoadSnapshotChunk   load_snapshot_chunk   = 14;
+    v1.RequestApplySnapshotChunk  apply_snapshot_chunk  = 15;
+    RequestPrepareProposal        prepare_proposal      = 16;
+    RequestProcessProposal        process_proposal      = 17;
+    RequestExtendVote             extend_vote           = 18;
+    RequestVerifyVoteExtension    verify_vote_extension = 19;
+    RequestFinalizeBlock          finalize_block        = 20;
+  }
+  reserved 4, 7, 9, 10;  // SetOption, BeginBlock, DeliverTx, EndBlock
+}
+
+message RequestInitChain {
+  google.protobuf.Timestamp time = 1
+      [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+  string                           chain_id         = 2;
+  cometbft.types.v3.ConsensusParams consensus_params = 3;
+  repeated v1.ValidatorUpdate      validators       = 4 [(gogoproto.nullable) = false];
+  bytes                            app_state_bytes  = 5;
+  int64                            initial_height   = 6;
+}
+
+message RequestPrepareProposal {
+  // the modified transactions cannot exceed this size.
+  int64 max_tx_bytes = 1;
+  // txs is an array of transactions that will be included in a block,
+  // sent to the app for possible modifications.
+  repeated bytes            txs                  = 2;
+  ExtendedCommitInfo        local_last_commit    = 3 [(gogoproto.nullable) = false];
+  repeated v2.Misbehavior   misbehavior          = 4 [(gogoproto.nullable) = false];
+  int64                     height               = 5;
+  google.protobuf.Timestamp time                 = 6 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+  bytes                     next_validators_hash = 7;
+  // address of the public key of the validator proposing the block.
+  bytes proposer_address = 8;
+}
+
+message RequestProcessProposal {
```

Comment:
This is because of `CommitInfo`.

Perhaps it's not easy to see at a glance when the field types are referenced from the same package. So it was `v2.CommitInfo` for `v2.RequestProcessProposal`, but in the proto it looks as though the field definition has not changed in `v3`. Maybe I should disambiguate all type references with the v-package prefix for clarity, even where it refers to the file's own package.